src/counts.jl

```diff
@@ -413,10 +442,27 @@ countmap(x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real} = addcoun
 
 
 """
-    proportionmap(x)
+    proportionmap(x; alg = :auto)
+    proportionmap(x::AbstractVector, w::AbstractVector{<:Real})
+
+Return a dictionary mapping each unique value in `x` to its proportion in `x`.
+
+If a vector of weights `wv` is provided, the proportion of weights is computed rather
+than the proportion of raw counts.
+
+`alg` is only allowed for unweighted counting and can be one of:
+- `:auto` (default): if `StatsBase.radixsort_safe(eltype(x)) == true` then use
+                     `:radixsort`, otherwise use `:dict`.
 
-Return a dictionary mapping each unique value in `x` to its
-proportion in `x`.
+- `:radixsort`:      if `radixsort_safe(eltype(x)) == true` then use the
+                     [radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+                     algorithm to sort the input vector which will generally lead to
+                     shorter running time. However the radix sort algorithm creates a
+                     copy of the input vector and hence uses more RAM. Choose `:dict`
+                     if the amount of available RAM is a limitation.
+
+- `:dict`:           use `Dict`-based method which is generally slower but uses less
+                     RAM and is safe for any data type.
 """
-proportionmap(x::AbstractArray) = _normalize_countmap(countmap(x), length(x))
+proportionmap(x::AbstractArray; alg = :auto) = _normalize_countmap(countmap(x; alg = alg), length(x))
```

Comment:
It's nice to add this argument, but can you add a test that uses it then? Or just leave this change out for now.

Reply:
Good catch! I tested on offset arrays but didn't test specified alg.

Reply:
I just left the change out